pkg/provision/storage/shared.go

```diff
@@ -277,3 +277,32 @@ func checkPVCTerminating(name, namespace string, api sync.ClusterAPI) (bool, err
 	}
 	return pvc.DeletionTimestamp != nil, nil
 }
+
+func shouldPersistHomeDirectory(workspace *common.DevWorkspaceWithConfig) bool {
+	return workspace.Config.Workspace.PersistUserHome != nil && workspace.Config.Workspace.PersistUserHome.Enabled != nil && *workspace.Config.Workspace.PersistUserHome.Enabled
+}
+
+// Adds a home volume mount to the given set of Pod Additions.
+// The home volume mount is mounted to the given PVC at the path `/home/user/`.
+// If a Devfile volume already mounts to `/home/user/`, an error is returned.
+func addHomeVolumeMount(workspace *common.DevWorkspaceWithConfig, pvcName string, podAdditions *v1alpha1.PodAdditions) error {
+	homeUserDirPath := "/home/user/"
+	homeVolumeMount := corev1.VolumeMount{
+		Name:      pvcName,
+		MountPath: homeUserDirPath,
+		SubPath:   fmt.Sprintf("%s/home/user/", workspace.Status.DevWorkspaceId),
```

Comment:
I've two concerns here:

* Defining the home mount in this way puts how volumeMounts are defined in two places -- one is in `(provisioner).rewriteVolumeMounts` and one is now here. StorageProvisioners are no longer have full control over how volumeMounts are handled
* This is using a different subpath scheme for per-workspace PVCs, which normally just use `volumeMount.Name` as the subpath (since they don't need to scope by workspace ID)

Reply:
> This is using a different subpath scheme for per-workspace PVCs, which normally just use volumeMount.Name as the subpath (since they don't need to scope by workspace ID)

Glad you pointed this out. I figured this would be an okay "oddity" in exchange for less code duplication. Will address this.